### DISCUSSION TITLE
How do we implement and Test P2P sync capabilities for FHIR resources and other related artifacts?

### DISCUSSION THREAD

**Discussion by f-odhiambo** [opensrp/fhircore#350] [Ideas]
We need to figure how p2p sync will work on FHIR Core and also the supported use cases for it. Also, try to address historical issues on Data loss

**Comment by pld** [comment]
Have we confirmed this will be in FHIR Core and not the Android FHIR SDK?

**Reply by f-odhiambo** [reply]
Yes. I was converting issues that were best suited as discussions. @pld . Will track this using a separate created on the SDK side

**Reply by pld** [reply]
K, this is the SDK issue, https://github.com/google/android-fhir/issues/258

**Comment by f-odhiambo** [comment]
P2P sync is an essential component of the Liberia CHW app. Given that CHWs are always offline, P2P sync is the only way to get data to the server. There is no viable solution in that environment without functional P2P sync. 

The P2P sync is primarily used on the monthly supervision visits conducted by the CHW supervisors (CHSSs). The CHSS is tasked with visiting each of their CHWs once a month. During this visit, they will bump all data off of their CHW’s device onto their own. Then, once a month, the CHSS will bump data to a district health officer (DHO) for them to sync to the server. 

1. CHW syncs data to their CHSS (CHSS has full permissions to view)
2. CHSS syncs data to a DHO (DHO has full permissions to view) 
3. CHW syncs data to another CHW (CHW cannot view data) 
4. CHW syncs their data + another CHW’s data to their shared CHSS (CHSS has full permissions to view) 
5. At the moment there is no real use case to sync data back down the chain of supervision

In our current implementation, we also allow CHWs to bump data between devices because there are use cases where one may not be able to reach a CHSS as reliably. In this case, a CHW will store that data on their device but not be able to interact with it (as the patients are not a part of their catchment). 

These workflows introduce some complexity in terms of permissions and editing. For example, in Liberia, we consider the data on the CHW’s device to be the source of truth. Therefore, CHSS’s, though they can access all patient data, are not to edit those records. And if they do, at sync, the CHW’s record will trump. We may explore solutions upstream via restrictions in the app, but it is important to note the potential impact of P2P sync on sync.

**Comment by f-odhiambo** [comment]
1. Some running notes [here](https://docs.google.com/document/d/1rdRu0_z3KN7fUUoqq9ZuVsUdhjtf70p13axwjT7ZRVU/edit#heading=h.t7nodeerl1nl) from previous check-in calls 
2. SDK Issue being tracked here issue [#258](https://github.com/google/android-fhir/issues/258)
3. Peer-to-peer UI/UX tracked here #465
4. Implementation ticket tracked here  : CHW/ANC - Allow for P2P Sync #281
